### PR TITLE
chore(OSSF): update branch protection to improve OSSF scorecard

### PR DESCRIPTION
**Pull request by Sabarivasan-Velayutham**
### Issue: [Bump OSSF Score Above 9.0 ⬆️](https://github.com/finos/git-proxy/issues/694)

image

This pull request focuses on improving the security configuration of our repository to enhance our OSSF Scorecard rating above 9.0. This involved  changes in  protections and access token usage.

---

#### Changes Implemented

1. **Enhanced Security Configurations**
   - Applied branch protection rules to the `main` branch to strengthen security, which includes:
     - **Required Pull Requests** for all changes on `main`.
     - **Dismiss Stale Reviews** to ensure each approval reflects the latest commit.
     - **Require Approvals** with a minimum of 2 reviewers.
     - **Up-to-Date Branches** must be maintained for pull requests targeting `main`.
     - **CODEOWNERS file** added in the `.github` folder to enforce code ownership and ensure critical files receive proper review before merging.
    
image

---

image

2. **Improved OSSF Scorecard Rating**
   - **Classic Token Usage**: To generate the necessary permissions for Scorecard evaluation, a classic token was created with `repo`,  `read:org` and `read:discussion` scopes. Fine-grained tokens may not provide the required permissions.
   - 
image
image

---

#### Benefits

- **Enhanced Security Posture**: By strictly scoping token permissions and enforcing protective branch rules, we mitigate the risk of unauthorized access or privilege misuse.
- **Scorecard Improvement**: Adopting these changes significantly boosts our OSSF Scorecard score, reflecting our commitment to security best practices.

---

#### Checklist

- ✅ Established critical branch protections for `main`.
- ✅ Created a `CODEOWNERS` file to enforce ownership and improve scorecard evaluation.

### CODE REVIEW

**Comment by Sabarivasan-Velayutham** [comment]
@JamieSlome I am a fresher working in Citi and I have sent an authorization request as a Corporate Contributor for EasyCLA. Can you please approve it?

**Comment by laukik-target** [comment]
> @JamieSlome I am a fresher working in Citi and I have sent an authorization request as a Corporate Contributor for EasyCLA. Can you please approve it?

Hey, better to contribute as an Individual. if your GitHub account is Personal then select individual.

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "client": "vite --config vite.config.js",
     "clientinstall": "npm install --prefix client",
     "server": "node index.js",
-    "start": "concurrently 'npm run server' 'npm run client'",
+    "start": "concurrently \"npm run server\" \"npm run client\"",
```
Comment (laukik-target) [inline package.json:10]: You can remove this, I have already raised PR for this here https://github.com/finos/git-proxy/pull/784

**Review by laukik-target — CHANGES_REQUESTED** [review]
1. Remove .env.development, package.json, package-lock.json
2. I don't see any change related to branch protection.

Please revise your change.

@JamieSlome  I think Branch Protection Settings can be updated by Maintainers only as we do not have access to update the settings.

File: package.json [hunk 2]
```diff
@@ -63,6 +63,7 @@
     "moment": "^2.29.4",
     "mongodb": "^5.0.0",
     "nodemailer": "^6.6.1",
+    "pack": "^2.2.0",
```
Comment (JamieSlome) [inline package.json:66]: What is the purpose of this new dependency?

File: client/package.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "@finos/git-proxy": "file:.."
+  }
+}
```
Comment (JamieSlome) [inline client/package.json:5]: Why are we making this change?

Comment (Sabarivasan-Velayutham) [reply client/package.json:5]: I installed dependencies for this project using the commands  `npm run clientinstall` ( which installed the above dependency ) and `npm install`. This dependency is not relatable for branch protection. 

Branch protection can be modified by making the above changes in branch settings which I have shown in the screenshots.

**Pull request closed without merging**